File: VB/MailMerge/SampleData.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.Collections

Namespace MailMerge
	Friend Class SampleData
		Inherits ArrayList
		Public Sub New()
			Add(New AddresseeRecord("Andrew", "XYZ Inc.", "Tullamore Way, 21", 36))
			Add(New AddresseeRecord("Benny", "ABC Ltd.", "Casablanca Rd., 46", 21))
			Add(New AddresseeRecord("Jose", "CASA S.A.", "Paseo di Ronda, 88", 45))


		End Sub
	End Class

	Public Class AddresseeRecord
		Private fAge As Integer
		Private fName As String
		Private fCompany As String
		Private fAddress As String

		Public Property Age() As Integer
			Get
				Return fAge
			End Get
			Set(ByVal value As Integer)
				fAge = value
			End Set
		End Property
		Public Property Name() As String
			Get
				Return fName
			End Get
			Set(ByVal value As String)
				fName = value
			End Set
		End Property
		Public Property Company() As String
			Get
				Return fCompany
			End Get
			Set(ByVal value As String)
				fCompany = value
			End Set
		End Property
		Public Property Address() As String
			Get
				Return fAddress
			End Get
			Set(ByVal value As String)
				fAddress = value
			End Set
		End Property

		Public Sub New(ByVal fName As String, ByVal fCompany As String, ByVal fAddress As String, ByVal fAge As Integer)
			Me.fName = fName
			Me.fAge = fAge
			Me.fCompany = fCompany
			Me.fAddress = fAddress
		End Sub
	End Class
End Namespace

File: CS/MailMerge/SampleData.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Collections;

namespace MailMerge {
    class SampleData : ArrayList {
        public SampleData() {
            Add(new AddresseeRecord("Andrew", "XYZ Inc.", "Tullamore Way, 21", 36));
            Add(new AddresseeRecord("Benny", "ABC Ltd.", "Casablanca Rd., 46", 21));
            Add(new AddresseeRecord("Jose", "CASA S.A.", "Paseo di Ronda, 88", 45));


        }
    }

    public class AddresseeRecord {
        private int fAge;
        private string fName;
        private string fCompany;
        private string fAddress;

        public int Age {
            get { return fAge; }
            set { fAge = value; }
        }
        public string Name {
            get { return fName; }
            set { fName = value; }
        }
        public string Company {
            get { return fCompany; }
            set { fCompany = value; }
        }
        public string Address {
            get { return fAddress; }
            set { fAddress = value; }
        }

        public AddresseeRecord(string fName, string fCompany, string fAddress, int fAge) {
            this.fName = fName;
            this.fAge = fAge;
            this.fCompany = fCompany;
            this.fAddress = fAddress;
        }
    }
}
